test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Smart/SmartSearchTests.cs

```diff
@@ -0,0 +1,358 @@
+                _fhirStorageTestHelper = _fixture.TestHelper;
+                _scopedDataStore = _fixture.DataStore.CreateMockScope();
+
+                _searchParameterDefinitionManager = _fixture.SearchParameterDefinitionManager;
+                _supportedSearchParameterDefinitionManager = _fixture.SupportedSearchParameterDefinitionManager;
+
+                _typedElementToSearchValueConverterManager = await CreateFhirTypedElementToSearchValueConverterManagerAsync();
+
+                _searchIndexer = new TypedElementSearchIndexer(
+                    _supportedSearchParameterDefinitionManager,
+                    _typedElementToSearchValueConverterManager,
+                    Substitute.For<IReferenceToElementResolver>(),
+                    ModelInfoProvider.Instance,
+                    NullLogger<TypedElementSearchIndexer>.Instance);
+
+                ResourceWrapperFactory wrapperFactory = Mock.TypeWithArguments<ResourceWrapperFactory>(
+                    new RawResourceFactory(new FhirJsonSerializer()),
+                    new FhirRequestContextAccessor(),
+                    _searchIndexer,
+                    _searchParameterDefinitionManager,
+                    Deserializers.ResourceDeserializer);
+
+                _searchParameterStatusManager = _fixture.SearchParameterStatusManager;
+
+                _searchService = _fixture.SearchService.CreateMockScope();
+
+                _contextAccessor = _fixture.FhirRequestContextAccessor;
+
+                var smartBundle = Samples.GetJsonSample<Bundle>("SmartPatientA");
+                foreach (var entry in smartBundle.Entry)
+                {
+                    await PutResource(entry.Resource);
+                }
+            }
+        }
+
+        public Task DisposeAsync()
+        {
+            return Task.CompletedTask;
+        }
+
+        [SkippableFact]
+        public async Task GivenScopesWithReadForAllResources_WhenRevIncludeObservations_PatientAndObservationReturned()
+        {
+            Skip.If(
+                ModelInfoProvider.Instance.Version != FhirSpecification.R4 &&
+                ModelInfoProvider.Instance.Version != FhirSpecification.R4B,
+                "This test is only valid for R4 and R4B");
+
+            // try to query both the patient and the observation using revinclude
+            var query = new List<Tuple<string, string>>();
+            query.Add(new Tuple<string, string>("_id", "smart-patient-A"));
+            query.Add(new Tuple<string, string>("_revinclude", "Observation:subject"));
+
+            var scopeRestriction = new ScopeRestriction("*", Core.Features.Security.DataActions.Read, "patient");
+
+            ConfigureFhirRequestContext(_contextAccessor, new List<ScopeRestriction>() { scopeRestriction });
+
+            var results = await _searchService.Value.SearchAsync("Patient", query, CancellationToken.None);
+
+            // assert that only the patient is returned
```

Comment:
Wrong comment? Isn't this ensuring that Patient and Observation resources are returned, and not 'only the patient is returned'?

Reply:
Wrong comment, updated.

Reply:
I don't think this particular comment was updated, maybe the changes are not showing on my side or were left out?